### PR TITLE
[feat] Add Accelerate SFT Trainer

### PR DESCRIPTION
**Pull request by reciprocated**
This PR gathers supervised finetuning from these two places[1][2] under the shared api

[1] https://github.com/CarperAI/trlx/blob/main/examples/summarize_rlhf/sft/train_gptj_summarize.py
[2] https://github.com/Dahoas/reward-modeling/blob/main/reward-modeling/finetune_base.py

https://wandb.ai/sorry/trlx/reports/Add-Accelerate-SFT-Trainer-280--VmlldzozNTAyNzQ1

### CODE REVIEW

**Comment by cat-state** [comment]
Thanks @reciprocated , this mostly LGTM, could you update the other examples to also match the `samples, rewards` args instead of `dataset`?

**Comment by reciprocated** [comment]
@cat-state, I've updated those, but perhaps I was overzealous with making `TRLConfig` a positional and required argument, since the only example that doesn't work is `simulacra.py` and by extension also simulacra's colab which git cloned main, since it didn't provide a config of its own. Should I maybe revert those changes? Not sure how much have you progressed with replacing the current configuration, and how much the above still matters

File: trlx/trlx.py [hunk 1]
```diff
@@ -1,111 +1,117 @@
...
 
 from trlx.data.configs import TRLConfig
 from trlx.utils import set_seed
 from trlx.utils.loading import get_orchestrator, get_pipeline, get_trainer
 
 
-def train(
+def train(  # noqa: C901
+    config: Optional[TRLConfig] = None,
     model_path: Optional[str] = None,
```
Comment (cat-state) [inline trlx/trlx.py:12]: Maybe model path should still be the first arg? It's also whats still in the readme.

**Comment by cat-state** [comment]
For the future config changes, we'll still need to keep the ModelConfig part around so keeping `model_path` makes sense for now. We can update the simulacra colab to give a model_path which should make this work?

File: trlx/trlx.py (hunk 1 above)
Comment (reciprocated) [reply trlx/trlx.py:11]: yeah, reverted that bit

**Review by cat-state — APPROVED** [review]
okay! This LGTM now

**Pull request merged**